src/repo/mem.rs

```diff
@@ -61,6 +63,18 @@ impl BlockStore for MemBlockStore {
         self.blocks.lock().await.remove(cid);
         Ok(())
     }
+
+    async fn list(&self) -> Result<Vec<Result<Cid, Error>>, Error> {
+        let blocks_mutex = self.blocks.lock().await;
```

Comment:
```suggestion
        let guard = self.blocks.lock().await;
```

This is the more common name. `blocks` would be fine as well, but calling it a mutex when it's not doesn't seem right.